src/vmm/src/guest_config/mod.rs

```diff
@@ -1,5 +1,54 @@
 // Copyright 2023 Amazon.com, Inc. or its affiliates. All Rights Reserved.
 // SPDX-License-Identifier: Apache-2.0
 
+/// Contains common types for vmm to use internally
+/// as well as by user interface related code.
+pub mod common;
 #[cfg(cpuid)]
 pub mod cpuid;
+
+// TODO Remove allow(unused) once implementation added
+/// Module containing type implementations needed for x86 CPU configuration
+#[allow(unused)]
+#[cfg(target_arch = "x86_64")]
+pub mod x86_config {
+    use std::collections::HashSet;
+
+    use crate::guest_config::common::{CPUConfiguration, CPUTemplate, ConfigurationModifier};
+
+    impl ConfigurationModifier for CPUConfiguration {
+        fn apply_template(&self, cpu_template: CPUTemplate) -> Box<CPUConfiguration> {
+            // TODO - Apply template
+            get_host_cpu_configuration()
+        }
+    }
+
+    fn get_host_cpu_configuration() -> Box<CPUConfiguration> {
+        // TODO - Retrieve host configuration
+
+        Box::new(CPUConfiguration::new(HashSet::new()))
+    }
+}
+
+// TODO Remove allow(unused) once implementation added
+/// Module containing type implementations needed for aarch64 (ARM) CPU configuration
+#[allow(unused)]
+#[cfg(target_arch = "aarch64")]
+pub mod aarch64_config {
+    use std::collections::HashSet;
+
+    use crate::guest_config::common::{CPUConfiguration, CPUTemplate, ConfigurationModifier};
+
+    impl ConfigurationModifier for CPUConfiguration {
+        fn apply_template(&self, cpu_template: CPUTemplate) -> Box<CPUConfiguration> {
+            // TODO - Apply template
+            get_host_cpu_configuration()
+        }
+    }
+
+    fn get_host_cpu_configuration() -> Box<CPUConfiguration> {
+        // TODO - Retrieve host configuration
+
+        Box::new(CPUConfiguration::new(HashSet::new()))
```

Comment:
In ARM case, `get_host_cpu_configuration()` depends on the register IDs in the CPU template, as we can't query all possible registers. This isn't a part of the common type definitions though, as I understand.

Reply:
This is indeed within the aarch64 specific module. This will require parameterization as well to filter on registers to be queried via KVM in any event, so I expect the definition to change somewhat anyway.